doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+for python)
+
+
+
+\   .. code-block:: cpp
+
+Rendered as
+
+    .. code-block:: cpp
+
+       int main()
+       {
+         cout << "Hello, World!";
+         return 0;
+       }
+
+
+For input/output examples use the following syntax
+
+::
+
+   >>> input begins with 3, right-angled brackets
+   output is directly below input without any brackets
+
+
+Directives
+~~~~~~~~~~~
+
+This is not a complete guide to the directives of reStructuredText. For more
+options see `Sphinx directives <https://www.sphinx-doc.org/en/master/usage/
+restructuredtext/directives.html>`_ or `Docutils section on directives
+<http://docutils.sourceforge.net/docs/ref/rst/directives.html>`_.
+
+Equations
+^^^^^^^^^
+
+For equations, use LaTeX markup:
+
+::
+
+    .. math::
+
+            f(x) = \int_{-\infty}^{\infty} \hat f(\xi)\ e^{2 \pi i x \xi}\,d\xi,
+
+Rendered as
+
+
+    .. math::
+
+            f(x) = \int_{-\infty}^{\infty} \hat f(\xi)\ e^{2 \pi i x \xi}\,d\xi,
+
+
+Admonitions
+^^^^^^^^^^^
+
+Use admonitions to draw readers attention to a particular point. Possible
+admonitions types include
+
+"see also", "attention", "caution", "danger", "error", "hint", "important",
+"note", "tip", "warning", "admonition"
+
+If you want a custom admonition use
+
+::
+
+   .. admonition:: custom name
+
+         Here is some text
+
+Rendered as
+
+
+   .. admonition:: custom name
+
+         Here is some text
+
+
+Roles
+~~~~~
+
+See `the roles section in the Sphinx docs <https://www.sphinx-doc.org/en/
+master/usage/restructuredtext/roles.html>`_ for details.
+
+In-text math
+^^^^^^^^^^^^
+
+For in-text math, use
+
+::
+
+   Now we can see :math:`x=1` for this example.
+
+This will be rendered as
+
+
+   Now we can see :math:`x=1` for this example.
+
+Referencing
+^^^^^^^^^^^
+
+For referencing reStructuredText files in repository, use
+
+::
+
+   :doc:`file`
+
+::
+
+   :doc:`custom name <path/file>`
+
+This will be rendered as
+
+   :doc:`file`
+
+   :doc:`custom name <file>`
+
+
+For section headings, you can use either of the two examples below. But to
+reference figures or arbitrary places in a file, you must include a custom
+name (second example) in the reference for it to work.
+
+::
+
+    :ref:`ref_name`
+
+or
+
+::
+
+    :ref:`custom name <ref_name>`
+
+Rendered as
+
+
+    :ref:`ref_name`
+
+    :ref:`custom name <ref_name>`
+
+
+.. note::
+
+    The `ref_name` needs to be above the section you want to reference with the
+    following syntax
+
+    ::
+
+        .. _ref_name:
+
+        secton_header
+        --------------
+
+Special links
+^^^^^^^^^^^^^
+
+.. attention::
+
+  The items in this section are still in development and have not been incorporated into nest:master!
+
+
+To link PyNEST API functions used in-text to the API reference page use the following syntax:
+
+::
+
+   :py:func:`.Create`
+
+
+Rendered as
+
+   :py:func:`.Create`
+
+
+To link terms to the glossary page use the HoverXTooltip role from Mahdi Enan (INM-6)
```

Comment:
Can we link to somewhere from the name? Maybe a GitHub repository or such?

Reply:
Is this role available from a general repository for Sphinx/RST resources? Does it need to be installed? If it is not available from general source, I think we should include the necessary code in the NEST repo to avoid dependencies on personal repos.

Reply:
Or under the NEST or INM-6 account/organisation, so it can be used by NESTML as well!

Reply:
@jougs @heplesser @clinssen Mahdi developed this Sphinx extension in the INM-6 repo: https://github.com/INM-6/HoverXTooltip, It was put there as recommended by @terhorstd, since he knows more about the regulations and best practices to this kind of thing. I can add a link to the repo if that is appropriate.

Until this Hovertooltip is discussed and merged, same with the links to functions and intersphinx, I'd want to remove these  from the styleguide. I put them here now in case their PRs are ready while this one is still open.